#**DATA MATH 361 Lab Assignment 9** by John Dominguez-Trujillo

*ALL CODE WAS DEVELOPED IN VISUAL STUDIO CODE AND PASTED INTO GOOGLE COLAB*

###**Activity 1:** Find a data set with two quantitative variables and at least 50 observations. It can be the same one as in Lab 8. Write Python code to give a regression table. Use the table to identify the equation of the linear model. Use the table to decide whether the response variable is changing significantly as a function of the response variable. Justify your answer using three different numbers from the table. Be specific.


In [1]:
### Activity 1

## IMPORTS ESSENTIAL LIBRARIES
## PANDAS IS A LIBRARY USED FOR DATA MANIPULATION, ANALYSIS, AND DATA STRUCTURES
## STATSMODELS IS A LIBRARY USED FOR ESTIMATING STATISTICAL MODELS AND PERFORMING REGRESSION ANALYSIS, INCLUDING GENERATING REGRESSION TABLES
import pandas as pd
import statsmodels.api as sm

## LOADS DATA FROM CSV INTO A VARIABLE
## USING A URL THAT LINKS DIRECTLY TO THE CSV FILE IN MY GITHUB SO ANYONE CAN ACCESS IT
url = 'https://raw.githubusercontent.com/JohnDT-MechE/DATA-MATH-361-Lab-Assignments/main/Lab_Assignment_8/co2-emissions-vs-gdp.csv'
data = pd.read_csv(url)

## FILTERS THE DATA FOR THE YEAR 2019
data_2019 = data[data['Year'] == 2019]

## DROPS ROWS WITH MISSING VALUES IN 'ANNUAL CO₂ EMISSIONS (PER CAPITA)' AND 'GDP PER CAPITA' COLUMNS
data_2019 = data_2019.dropna(subset=['GDP per capita', 'Annual CO₂ emissions (per capita)'])

## EXTRACTS THE INDEPENDENT VARIABLE (GDP PER CAPITA) INTO X
## EXTRACTS THE DEPENDENT VARIABLE (ANNUAL CO₂ EMISSIONS (PER CAPITA)) INTO Y
X = data_2019['GDP per capita']
y = data_2019['Annual CO₂ emissions (per capita)']

## ADDS A COLUMN OF ONES TO X SO THE REGRESSION MODEL INCLUDES AN INTERCEPT TERM
X = sm.add_constant(X)

## SM.OLS FITS A LINEAR REGRESSION MODEL USING ORDINARY LEAST SQUARES
## .FIT() TRAINS THE MODEL ON THE DATA
model = sm.OLS(y, X).fit()

## PRINTS THE FULL REGRESSION TABLE TO THE OUTPUT TERMINAL
print(model.summary())

                                    OLS Regression Results                                   
Dep. Variable:     Annual CO₂ emissions (per capita)   R-squared:                       0.621
Model:                                           OLS   Adj. R-squared:                  0.619
Method:                                Least Squares   F-statistic:                     267.0
Date:                               Sun, 06 Apr 2025   Prob (F-statistic):           3.68e-36
Time:                                       22:07:00   Log-Likelihood:                -442.12
No. Observations:                                165   AIC:                             888.2
Df Residuals:                                    163   BIC:                             894.4
Df Model:                                          1                                         
Covariance Type:                           nonrobust                                         
                     coef    std err          t      P>|t|  

Yes, the response variable, the annual $CO_2$ emmisions per capita, is changing **significantly** as a function of the explanatory variable, the GDP per capita. This conclusion can be made due to three specific values from the regression tables. First, the p-value for the GDP per capita coefficient is very small (equal to 0.000) and is far below the standard significance level of 0.05. This means that the relationship between the GDP per capita and the annual $CO_2$ emissions per capita is statistically significant. It also indicates that there is strong evidence against the null hypothesis that the slppe is zero. Due to this, it can be confidently said that there is a linear relationship. Second, the slope of the regression line is about 0.0002. This means that for every \$1 increase in GDP per capita, the annual $CO_2$ emmisions per capita increase by 0.0002 metric tons. Since this is a positive value, it indicates that there is a direct relationship between the two variables. Finally, the $R^2$ value is about 0.621. This means that about 62.1% of the variation in annual $CO_2$ emissions per capita can be explained by GDP per capita. It also indicates that the model fits the data reasonably well. When put together, these values show that the model is statistically significant and that the GDP per capita is a meaningful predictor of annual $CO_2$ emmissionas per capita.